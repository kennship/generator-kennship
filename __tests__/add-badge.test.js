const addBadge = require('../generators/util/add-badge');

describe('util/addBadge', () => {
  it('should add a badge', async () => {
    const markdown = `# generator-kennship

> Common tasks for new projects

[![NPM version][npm-image]][npm-url]
[![Build Status][travis-image]][travis-url]
[![Coverage percentage][coveralls-image]][coveralls-url]

More content

[npm-image]: https://badge.fury.io/js/generator-kennship.svg
[npm-url]: https://npmjs.org/package/generator-kennship
[coveralls-image]: https://coveralls.io/repos/ryaninvents/generator-kennship/badge.svg
[coveralls-url]: https://coveralls.io/r/ryaninvents/generator-kennship
`;
    const withBadge = await addBadge(markdown, {
      name: 'github-stars',
      altText: 'GitHub stars',
      imageUrl:
        'https://img.shields.io/github/stars/kennship/generator-kennship.svg?style=social',
      href: 'https://github.com/kennship/generator-kennship',
      priority: 25,
    });

    expect(withBadge).toMatchSnapshot();
  });
});
