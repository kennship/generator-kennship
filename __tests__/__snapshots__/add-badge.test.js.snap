// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/addBadge should add a badge 1`] = `
"# generator-kennship

> Common tasks for new projects

[![NPM version][npm-image]][npm-url]
[![Build Status][travis-image]][travis-url]
[![GitHub stars][github-stars-image]][github-stars-url]
[![Coverage percentage][coveralls-image]][coveralls-url]

More content

[npm-image]: https://badge.fury.io/js/generator-kennship.svg

[npm-url]: https://npmjs.org/package/generator-kennship

[coveralls-image]: https://coveralls.io/repos/ryaninvents/generator-kennship/badge.svg

[coveralls-url]: https://coveralls.io/r/ryaninvents/generator-kennship

[github-stars-image]: https://img.shields.io/github/stars/kennship/generator-kennship.svg?style=social

[github-stars-url]: https://github.com/kennship/generator-kennship
"
`;

exports[`util/addBadge should behave well in a mostly-empty README 1`] = `
"# generator-kennship

> Common tasks for new projects

[![GitHub stars][github-stars-image]][github-stars-url]

## Content

Barebones README

[github-stars-image]: https://img.shields.io/github/stars/kennship/generator-kennship.svg?style=social

[github-stars-url]: https://github.com/kennship/generator-kennship
"
`;
